### DISCUSSION TITLE
How to avoid deleting ignored files in version control with gitignore

### DISCUSSION THREAD

**Discussion by incloud02** [community/community#118406] [New to GitHub]
I have added some items in Gitignore, but those items will disappear when I make version changes. I hope they will remain the same as before I made the version change.

**Comment by Pranav-Karra-3301** [comment]
It sounds like you're experiencing an issue where files that you've added to your `.gitignore` are disappearing when you switch versions. If the files are already being tracked by Git (i.e., they were committed before you added them to `.gitignore`), adding them to `.gitignore` won't affect their tracking status.

Here's what you can do to stop tracking these files without deleting them from your local file system:

1. Make sure you commit any outstanding changes.
2. Use the command `git rm --cached <file>` to remove the files from tracking, but keep them in your working directory.
3. Commit this change.
4. The files should now be ignored for future commits due to your `.gitignore`.

Remember to do this for each file or pattern you've added to `.gitignore` that was previously committed to your repository.

I hope this helps! If you have any further questions or run into any issues, feel free to ask.
